### ISSUE TITLE
ServiceManager not found in the ServiceLocator

### ISSUE THREAD

**Issue by Michal-sk** [zfcampus/zf-oauth2#131]
Updated today and now getting the following issue:
```Zend\ServiceManager\ServiceManager::get was unable to fetch or create an instance for ServiceManager```

```
<pre class="prettyprint linenums">
#0 C:\foo\kp-webservice\vendor\zfcampus\zf-oauth2\src\Factory\AuthControllerFactory.php(22): Zend\ServiceManager\ServiceManager->get('ServiceManager')
#1 [internal function]: ZF\OAuth2\Factory\AuthControllerFactory->createService(Object(Zend\Mvc\Controller\ControllerManager), 'zfoauth2control...', 'ZF\\OAuth2\\Contr...')
#2 C:\foo\kp-webservice\vendor\zendframework\zend-servicemanager\src\ServiceManager.php(936): call_user_func(Array, Object(Zend\Mvc\Controller\ControllerManager), 'zfoauth2control...', 'ZF\\OAuth2\\Contr...')
#3 C:\foo\kp-webservice\vendor\zendframework\zend-servicemanager\src\AbstractPluginManager.php(330): Zend\ServiceManager\ServiceManager->createServiceViaCallback(Array, 'zfoauth2control...', 'ZF\\OAuth2\\Contr...')
#4 C:\foo\kp-webservice\vendor\zendframework\zend-servicemanager\src\AbstractPluginManager.php(287): Zend\ServiceManager\AbstractPluginManager->createServiceViaCallback(Array, 'zfoauth2control...', 'ZF\\OAuth2\\Contr...')
#5 C:\foo\kp-webservice\vendor\zendframework\zend-servicemanager\src\ServiceManager.php(636): Zend\ServiceManager\AbstractPluginManager->createFromFactory('zfoauth2control...', 'ZF\\OAuth2\\Contr...')
#6 C:\foo\kp-webservice\vendor\zendframework\zend-servicemanager\src\ServiceManager.php(599): Zend\ServiceManager\ServiceManager->doCreate('ZF\\OAuth2\\Contr...', 'zfoauth2control...')
#7 C:\foo\kp-webservice\vendor\zendframework\zend-servicemanager\src\ServiceManager.php(532): Zend\ServiceManager\ServiceManager->create(Array)
#8 C:\foo\kp-webservice\vendor\zendframework\zend-servicemanager\src\AbstractPluginManager.php(161): Zend\ServiceManager\ServiceManager->get('ZF\\OAuth2\\Contr...', true)
#9 C:\foo\kp-webservice\vendor\zendframework\zend-mvc\src\DispatchListener.php(94): Zend\ServiceManager\AbstractPluginManager->get('ZF\\OAuth2\\Contr...')
#10 [internal function]: Zend\Mvc\DispatchListener->onDispatch(Object(Zend\Mvc\MvcEvent))
#11 C:\foo\kp-webservice\vendor\zendframework\zend-eventmanager\src\EventManager.php(490): call_user_func(Array, Object(Zend\Mvc\MvcEvent))
#12 C:\foo\kp-webservice\vendor\zendframework\zend-eventmanager\src\EventManager.php(263): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#13 C:\foo\kp-webservice\vendor\zendframework\zend-mvc\src\Application.php(340): Zend\EventManager\EventManager->triggerEventUntil(Object(Closure), Object(Zend\Mvc\MvcEvent))
#14 C:\foo\kp-webservice\public\index.php(51): Zend\Mvc\Application->run()
#15 {main}
</pre>
```

**Comment by Michal-sk** [comment]
```
zendframework/zend-authentication           2.6.x-dev 1dbbd4b                         provides an API for authentication and includes concrete authentication adapters for common use case scenarios
zendframework/zend-barcode                  2.7.x-dev 051d496                         provides a generic way to generate barcodes
zendframework/zend-cache                    2.7.x-dev ac2b0c0                         provides a generic way to cache any data
zendframework/zend-captcha                  2.6.x-dev 7bc9e4f
zendframework/zend-code                     2.6.2                                     provides facilities to generate arbitrary code using an object oriented interface
zendframework/zend-config                   2.7.x-dev 0f9a97a                         provides a nested object property based user interface for accessing this configuration data within application code
zendframework/zend-console                  2.7.x-dev b4f9e68
zendframework/zend-crypt                    2.7.x-dev baf577a
zendframework/zend-db                       2.8.x-dev 79d2b6f
zendframework/zend-debug                    2.6.x-dev 15f3f56
zendframework/zend-developer-tools          dev-master b2509f8                        Module for developer and debug tools for working with the ZF2 MVC layer.
zendframework/zend-di                       2.7.x-dev 6cdab81
zendframework/zend-diactoros                1.4.x-dev c866b82                         PSR HTTP Message implementations
zendframework/zend-dom                      2.6.x-dev adfa3ce                         provides tools for working with DOM documents and structures
zendframework/zend-escaper                  2.6.x-dev 2c4c319
zendframework/zend-eventmanager             2.6.x-dev 3d41b61
zendframework/zend-feed                     2.8.x-dev 184d29c                         provides functionality for consuming RSS and Atom feeds
zendframework/zend-file                     2.7.x-dev 1474111
zendframework/zend-filter                   2.7.x-dev 5fcced1                         provides a set of commonly needed data filters
zendframework/zend-form                     2.8.x-dev 1a878d4
zendframework/zend-http                     2.6.x-dev e0a1ce2                         provides an easy interface for performing Hyper-Text Transfer Protocol (HTTP) requests
zendframework/zend-hydrator                 1.1.x-dev 22652e1
zendframework/zend-i18n                     2.7.x-dev 478ad30
zendframework/zend-i18n-resources           2.5.2                                     Provides validator translations for zend-i18n's Translator
zendframework/zend-inputfilter              2.7.x-dev 1145bf2
zendframework/zend-json                     2.7.x-dev 4a8695d                         provides convenience methods for serializing native PHP to JSON and decoding JSON to native PHP
zendframework/zend-loader                   2.6.x-dev 5b29591
zendframework/zend-log                      2.8.x-dev 9a619ec                         component for general purpose logging
zendframework/zend-mail                     2.7.x-dev ec3a56e                         provides generalized functionality to compose and send both text and MIME-compliant multipart e-mail messages
zendframework/zend-math                     2.7.x-dev 61b07e0
zendframework/zend-memory                   2.6.x-dev 6959302
zendframework/zend-mime                     2.6.x-dev 7408e02
zendframework/zend-modulemanager            2.8.x-dev 9eb957c
zendframework/zend-mvc                      2.7.x-dev 45d259c
zendframework/zend-navigation               2.7.x-dev 52b2eb7                         provides support for managing trees of pointers to web pages
zendframework/zend-paginator                2.7.x-dev b25615c
zendframework/zend-permissions-acl          2.7.x-dev 3264aa2                         provides a lightweight and flexible access control list (ACL) implementation for privileges management
zendframework/zend-permissions-rbac         2.6.x-dev 90eac91                         provides a role-based access control management
zendframework/zend-progressbar              2.6.x-dev de8afa6                         component to create and update progressbars in different environments
zendframework/zend-psr7bridge               0.2.1                                     PSR-7 <-> Zend\Http bridge
zendframework/zend-serializer               2.7.x-dev 686cf55                         provides an adapter based interface to simply generate storable representation of PHP types by different facilities, and recover
zendframework/zend-server                   2.7.x-dev 89023e2
zendframework/zend-servicemanager           2.7.5
zendframework/zend-session                  2.7.x-dev 1db2f14                         manage and preserve session data, a logical complement of cookie data, across multiple page requests by the same client
zendframework/zend-soap                     2.6.x-dev 1b26b30
zendframework/zend-stdlib                   2.7.x-dev 4499760
zendframework/zend-tag                      2.7.x-dev 32d7173                         a component suite which provides a facility to work with taggable Items
zendframework/zend-test                     2.6.x-dev ff0bf89
zendframework/zend-text                     2.7.x-dev ae3138b
zendframework/zend-uri                      2.6.x-dev 75bcb74                         a component that aids in manipulating and validating ┬╗ Uniform Resource Identifiers (URIs)
zendframework/zend-validator                2.7.x-dev a48f5b2                         provides a set of commonly needed validators
zendframework/zend-version                  2.6.x-dev a74123d
zendframework/zend-view                     2.7.x-dev 537a33d                         provides a system of helpers, output filters, and variable escaping
zendframework/zend-xmlrpc                   2.6.x-dev 939989f
zendframework/zenddiagnostics               dev-master 2eeb413                        A set of components for performing diagnostic tests in PHP applications
zendframework/zendframework                 2.6.x-dev 17c0540                         Zend Framework 2
zendframework/zendxml                       dev-master 185bbf4                        Utility library for XML usage, best practices, and security in PHP
zendframework/zftool                        dev-master 840c3ee                        Utility module for Zend Framework 2 applications.
zfcampus/zf-api-problem                     1.2.x-dev ba209e6                         ZF2 Module providing API-Problem assets and rendering
zfcampus/zf-apigility                       1.3.x-dev 1df863f                         Apigility module for ZF2
zfcampus/zf-apigility-admin                 1.5.x-dev c5d664f                         Apigility Admin module
zfcampus/zf-apigility-admin-ui              dev-master 8084dd9                        Apigility Admin UI module
zfcampus/zf-apigility-documentation         1.2.x-dev 9f97cd1                         Apigility API documentation module
zfcampus/zf-apigility-documentation-swagger 1.2.x-dev 5dab6e8                         Apigility API Swagger documentation module
zfcampus/zf-apigility-provider              1.2.x-dev 079fcb7                         Apigility interfaces
zfcampus/zf-apigility-welcome               1.1.x-dev a7b16a7                         Apigility Welcome Screen
zfcampus/zf-configuration                   1.2.x-dev 5e55c8c                         ZF2 Module providing REST resource for manipulating configuration
zfcampus/zf-console                         1.3.x-dev 07f6e57                         Library for creating and dispatching console commands
zfcampus/zf-content-negotiation             dev-hotfix-array-data 42d9cb0             ZF2 Module providing content-negotiation features
zfcampus/zf-content-validation              1.3.x-dev 47655de                         ZF2 Module providing incoming content validation
zfcampus/zf-deploy                          1.2.x-dev 2d0a2b4                         Deployment tool for ZF2 application
zfcampus/zf-development-mode                2.2.x-dev d8f1496                         ZF2 development mode module
zfcampus/zf-hal                             1.4.x-dev ca780db                         ZF2 Module providing Hypermedia Application Language assets and rendering
zfcampus/zf-mvc-auth                        dev-hotfix-allow-override-adapter 9406f79 ZF2 Module providing Authentication and Authorization events and infrastructure
zfcampus/zf-oauth2                          1.4.x-dev 7fd1fda                         ZF2 module for implementing an OAuth2 server
zfcampus/zf-rest                            1.3.x-dev 764882e                         ZF2 Module providing structure for RESTful resources
zfcampus/zf-rpc                             1.3.x-dev 864e830                         ZF2 Module for simplifying the creation of RPC services
zfcampus/zf-versioning                      1.2.x-dev 4b21279                         ZF2 Module providing listeners and route prototypes for implementing API versioning

```

**Comment by Michal-sk** [comment]
Removed everything, reinstalled now it works... so just closing, altough unsure.

**Comment by Michal-sk** [comment]
And it's back. re-opening.

**Comment by metanav** [comment]
I am also facing same issue after upgrade. it's BC.

**Comment by Michal-sk** [comment]
for now in composer just use:
```json
    "require": {
      "zfcampus/zf-oauth2": "dev-iss131_servicemanager as 1.3.1"
    },
```

**Comment by metanav** [comment]
@Michal-sk I think the fix is not backward compatible. Can you confirm what changes has broken this?

**Comment by jguittard** [comment]
Fixed by #122
